hs_core/models.py

```diff
@@ -1962,6 +1962,34 @@ def get_irods_storage(self):
         else:
             return IrodsStorage()
 
+    def set_quota_holder(self, setter, new_holder):
+        # set quota holder of the resource to new_holder who must be an owner
+        # setter is the requesting user to transfer quota holder and setter must also be an owner
+        if not setter.uaccess.owns_resource(self) or not new_holder.uaccess.owns_resource(self):
+            raise PermissionDenied("Only owners can set or be set as quota holder for the resource")
+        istorage = self.get_irods_storage()
+        istorage.setAVU(self.root_path, "quotaUserName", new_holder.username)
+
+    def get_quota_holder(self):
+        # get quota holder of the resource
+        # return User instance of the quota holder for the resource or None if it does not exist
+        istorage = self.get_irods_storage()
+        try:
+            uname = istorage.getAVU(self.root_path, "quotaUserName")
+        except SessionException:
+            # quotaUserName AVU does not exist, return None
+            return None
+
+        if uname:
+            ufilter = User.objects.filter(username=uname)
```

Comment:
Hmm...not sure why I got the exception initially then, but agreed per the documentation first() call is sufficient without need to check exists() first. Will make the change then, and keep this in mind if things break under certain circumstances in the future.